pandas/tests/series/test_replace.py

```diff
@@ -305,3 +305,16 @@ def test_replace_with_no_overflowerror(self):
         result = s.replace(["100000000000000000000"], [1])
         expected = pd.Series([0, 1, "100000000000000000001"])
         tm.assert_series_equal(result, expected)
+
+    @pytest.mark.parametrize(
+        "ser, to_replace, value, exp",
+        [([1, 2, 3], 2, True, [1, True, 3]), (["x", 2, 3], 2, True, ["x", True, 3])],
+    )
+    def test_replace_no_cast(self, ser, to_replace, value, exp):
+        # GH 9113
+        # BUG: replace int64 dtype with bool coerces to int64
+
+        series = pd.Series(ser)
+        result = series.replace(to_replace, value)
```

Comment:
to_replace and value are the same for the cases, so don't make these args, just put it in the code